ingestion-beam/src/main/java/com/mozilla/telemetry/io/Write.java

```diff
@@ -276,18 +268,17 @@ public void processElement(ProcessContext ctx) {
 
       FileIO.Write<List<String>, PubsubMessage> write = FileIO
           .<List<String>, PubsubMessage>writeDynamic() //
-          .by(message -> pathTemplate.get().extractValuesFrom(message.getAttributeMap()))
+          .by(message -> pathTemplate.extractValuesFrom(message.getAttributeMap()))
           .withDestinationCoder(ListCoder.of(StringUtf8Coder.of())) //
           .withCompression(compression) //
           .via(Contextful.fn(encoder::getSink)) //
           .to(staticPrefix) //
           .withNaming(placeholderValues -> NoColonFileNaming
-              .defaultNaming(pathTemplate.get().replaceDynamicPart(placeholderValues), ".avro"));
+              .defaultNaming(pathTemplate.replaceDynamicPart(placeholderValues), ".avro"));
 
       if (inputType == InputType.pubsub) {
-        // Passing a ValueProvider to withNumShards disables runner-determined sharding, so we
-        // need to be careful to pass this only for streaming input (where runner-determined
-        // sharding is not an option).
+        // withNumShards disables runner-determined sharding, so we need to be careful to pass this
+        // only for streaming input (where runner-determined sharding is not an option).
         write = write.withNumShards(numShards);
       }
```

Comment:
unsure if condition is necessary here, too

Reply:
I think this conditional still makes sense. We should always use the default runner-determined sharding when in batch mode.